### ISSUE TITLE
Wrong inference result after running fuse_conv_bn.py

### ISSUE THREAD

**Issue by NygmaLin** [open-mmlab/mmdetection#2924]
Hello, I train my model with mmdet2.0, but I got the different and wrong inference results after running fuse_conv_bn.py without  any error or warning. When i ran my inference script,the waring log report as below:
The model and loaded state dict do not match exactly

unexpected key in source state_dict: backbone.conv1.bias, backbone.layer1.0.conv1.bias, backbone.layer1.0.conv2.bias, backbone.layer1.0.conv3.bias, backbone.layer1.0.downsample.0.bias, backbone.layer1.1.conv1.bias, backbone.layer1.1.conv2.bias, backbone.layer1.1.conv3.bias, backbone.layer1.2.conv1.bias, backbone.layer1.2.conv2.bias, backbone.layer1.2.conv3.bias, backbone.layer2.0.conv1.bias, backbone.layer2.0.conv3.bias, backbone.layer2.0.downsample.0.bias, backbone.layer2.1.conv1.bias, backbone.layer2.1.conv3.bias, backbone.layer2.2.conv1.bias, backbone.layer2.2.conv3.bias, backbone.layer2.3.conv1.bias, backbone.layer2.3.conv3.bias, backbone.layer3.0.conv1.bias, backbone.layer3.0.conv3.bias, backbone.layer3.0.downsample.0.bias, backbone.layer3.1.conv1.bias, backbone.layer3.1.conv3.bias, backbone.layer3.2.conv1.bias, backbone.layer3.2.conv3.bias, backbone.layer3.3.conv1.bias, backbone.layer3.3.conv3.bias, backbone.layer3.4.conv1.bias, backbone.layer3.4.conv3.bias, backbone.layer3.5.conv1.bias, backbone.layer3.5.conv3.bias, backbone.layer4.0.conv1.bias, backbone.layer4.0.conv3.bias, backbone.layer4.0.downsample.0.bias, backbone.layer4.1.conv1.bias, backbone.layer4.1.conv3.bias, backbone.layer4.2.conv1.bias, backbone.layer4.2.conv3.bias

missing keys in source state_dict: backbone.bn1.weight, backbone.bn1.bias, backbone.bn1.running_mean, backbone.bn1.running_var, backbone.layer1.0.bn1.weight, backbone.layer1.0.bn1.bias, backbone.layer1.0.bn1.running_mean, backbone.layer1.0.bn1.running_var, backbone.layer1.0.bn2.weight, backbone.layer1.0.bn2.bias, backbone.layer1.0.bn2.running_mean, backbone.layer1.0.bn2.running_var, backbone.layer1.0.bn3.weight, backbone.layer1.0.bn3.bias, backbone.layer1.0.bn3.running_mean, backbone.layer1.0.bn3.running_var, backbone.layer1.0.downsample.1.weight, backbone.layer1.0.downsample.1.bias, backbone.layer1.0.downsample.1.running_mean, backbone.layer1.0.downsample.1.running_var, backbone.layer1.1.bn1.weight, backbone.layer1.1.bn1.bias, backbone.layer1.1.bn1.running_mean, backbone.layer1.1.bn1.running_var, backbone.layer1.1.bn2.weight, backbone.layer1.1.bn2.bias, backbone.layer1.1.bn2.running_mean, backbone.layer1.1.bn2.running_var, backbone.layer1.1.bn3.weight, backbone.layer1.1.bn3.bias, backbone.layer1.1.bn3.running_mean, backbone.layer1.1.bn3.running_var, backbone.layer1.2.bn1.weight, backbone.layer1.2.bn1.bias, backbone.layer1.2.bn1.running_mean, backbone.layer1.2.bn1.running_var, backbone.layer1.2.bn2.weight, backbone.layer1.2.bn2.bias, backbone.layer1.2.bn2.running_mean, backbone.layer1.2.bn2.running_var, backbone.layer1.2.bn3.weight, backbone.layer1.2.bn3.bias, backbone.layer1.2.bn3.running_mean, backbone.layer1.2.bn3.running_var, backbone.layer2.0.bn1.weight, backbone.layer2.0.bn1.bias, backbone.layer2.0.bn1.running_mean, backbone.layer2.0.bn1.running_var, backbone.layer2.0.bn3.weight, backbone.layer2.0.bn3.bias, backbone.layer2.0.bn3.running_mean, backbone.layer2.0.bn3.running_var, backbone.layer2.0.downsample.1.weight, backbone.layer2.0.downsample.1.bias, backbone.layer2.0.downsample.1.running_mean, backbone.layer2.0.downsample.1.running_var, backbone.layer2.1.bn1.weight, backbone.layer2.1.bn1.bias, backbone.layer2.1.bn1.running_mean, backbone.layer2.1.bn1.running_var, backbone.layer2.1.bn3.weight, backbone.layer2.1.bn3.bias, backbone.layer2.1.bn3.running_mean, backbone.layer2.1.bn3.running_var, backbone.layer2.2.bn1.weight, backbone.layer2.2.bn1.bias, backbone.layer2.2.bn1.running_mean, backbone.layer2.2.bn1.running_var, backbone.layer2.2.bn3.weight, backbone.layer2.2.bn3.bias, backbone.layer2.2.bn3.running_mean, backbone.layer2.2.bn3.running_var, backbone.layer2.3.bn1.weight, backbone.layer2.3.bn1.bias, backbone.layer2.3.bn1.running_mean, backbone.layer2.3.bn1.running_var, backbone.layer2.3.bn3.weight, backbone.layer2.3.bn3.bias, backbone.layer2.3.bn3.running_mean, backbone.layer2.3.bn3.running_var, backbone.layer3.0.bn1.weight, backbone.layer3.0.bn1.bias, backbone.layer3.0.bn1.running_mean, backbone.layer3.0.bn1.running_var, backbone.layer3.0.bn3.weight, backbone.layer3.0.bn3.bias, backbone.layer3.0.bn3.running_mean, backbone.layer3.0.bn3.running_var, backbone.layer3.0.downsample.1.weight, backbone.layer3.0.downsample.1.bias, backbone.layer3.0.downsample.1.running_mean, backbone.layer3.0.downsample.1.running_var, backbone.layer3.1.bn1.weight, backbone.layer3.1.bn1.bias, backbone.layer3.1.bn1.running_mean, backbone.layer3.1.bn1.running_var, backbone.layer3.1.bn3.weight, backbone.layer3.1.bn3.bias, backbone.layer3.1.bn3.running_mean, backbone.layer3.1.bn3.running_var, backbone.layer3.2.bn1.weight, backbone.layer3.2.bn1.bias, backbone.layer3.2.bn1.running_mean, backbone.layer3.2.bn1.running_var, backbone.layer3.2.bn3.weight, backbone.layer3.2.bn3.bias, backbone.layer3.2.bn3.running_mean, backbone.layer3.2.bn3.running_var, backbone.layer3.3.bn1.weight, backbone.layer3.3.bn1.bias, backbone.layer3.3.bn1.running_mean, backbone.layer3.3.bn1.running_var, backbone.layer3.3.bn3.weight, backbone.layer3.3.bn3.bias, backbone.layer3.3.bn3.running_mean, backbone.layer3.3.bn3.running_var, backbone.layer3.4.bn1.weight, backbone.layer3.4.bn1.bias, backbone.layer3.4.bn1.running_mean, backbone.layer3.4.bn1.running_var, backbone.layer3.4.bn3.weight, backbone.layer3.4.bn3.bias, backbone.layer3.4.bn3.running_mean, backbone.layer3.4.bn3.running_var, backbone.layer3.5.bn1.weight, backbone.layer3.5.bn1.bias, backbone.layer3.5.bn1.running_mean, backbone.layer3.5.bn1.running_var, backbone.layer3.5.bn3.weight, backbone.layer3.5.bn3.bias, backbone.layer3.5.bn3.running_mean, backbone.layer3.5.bn3.running_var, backbone.layer4.0.bn1.weight, backbone.layer4.0.bn1.bias, backbone.layer4.0.bn1.running_mean, backbone.layer4.0.bn1.running_var, backbone.layer4.0.bn3.weight, backbone.layer4.0.bn3.bias, backbone.layer4.0.bn3.running_mean, backbone.layer4.0.bn3.running_var, backbone.layer4.0.downsample.1.weight, backbone.layer4.0.downsample.1.bias, backbone.layer4.0.downsample.1.running_mean, backbone.layer4.0.downsample.1.running_var, backbone.layer4.1.bn1.weight, backbone.layer4.1.bn1.bias, backbone.layer4.1.bn1.running_mean, backbone.layer4.1.bn1.running_var, backbone.layer4.1.bn3.weight, backbone.layer4.1.bn3.bias, backbone.layer4.1.bn3.running_mean, backbone.layer4.1.bn3.running_var, backbone.layer4.2.bn1.weight, backbone.layer4.2.bn1.bias, backbone.layer4.2.bn1.running_mean, backbone.layer4.2.bn1.running_var, backbone.layer4.2.bn3.weight, backbone.layer4.2.bn3.bias, backbone.layer4.2.bn3.running_mean, backbone.layer4.2.bn3.running_var

/data/mm/mmdetection2.0/mmdet/apis/inference.py:40: UserWarning: Class names are not saved in the checkpoint's meta data, use COCO classes by default.

**Comment by NygmaLin** [comment]
@hellock @yhcao6  
hello, I debugged the inference script and i found this error mainly caused by the unexpected_keys. The bias of original conv2d module in model is:
 Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False), 
you can see the parameter bias = False, but after running fuse_conv_bn.py, the bias of bn is moved to bias of conv layer. So the bias of conv is not empty,which means the module should be modified as:  
Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=True).
Meanwhile, the bn of  fused model are removed. So some important data are lacked.I guess it might be the source of this bug. 
I'm tring to fix it,and I hope you can fix it as soon as possible.thx

**Comment by yhcao6** [comment]
Could you provide the config and model, I can't reproduce your error.

**Comment by NygmaLin** [comment]
@yhcao6 hello,I found that when building backbone,the bias in conv always be False even I comment out the norm_cfg in config file.So after running init_detector from mmdet.apis, the bias of conv always be False.

**Comment by yhcao6** [comment]
You can refer to the construction of ResNet here. https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/backbones/resnet.py#L156.
For example the bias: 
https://github.com/open-mmlab/mmdetection/blob/99a31d25b4d685da5ae868776a0483b80e8fb903/mmdet/models/backbones/resnet.py#L150-L156
https://github.com/open-mmlab/mmdetection/blob/99a31d25b4d685da5ae868776a0483b80e8fb903/mmdet/models/backbones/resnet.py#L184-L189
